pytest_fixtures/component/rh_cloud.py

```diff
@@ -82,23 +83,28 @@ def rhcloud_registered_hosts(organization_ak_setup, content_hosts, rhcloud_sat_h
     return content_hosts
 
 
-@pytest.fixture(scope='module')
-def rhel8_insights_vm(rhcloud_sat_host, organization_ak_setup, rhel8_contenthost_module):
+@pytest.fixture
+def rhel_insights_vm(rhcloud_sat_host, organization_ak_setup, rhel_contenthost):
     """A module-level fixture to create rhel8 content host registered with insights."""
+    distro_repo_map = {
+        7: DISTRO_RHEL7,
+        8: DISTRO_RHEL8,
+        9: DISTRO_RHEL9,
+    }
     org, ak = organization_ak_setup
-    rhel8_contenthost_module.configure_rex(satellite=rhcloud_sat_host, org=org, register=False)
-    rhel8_contenthost_module.configure_rhai_client(
-        satellite=rhcloud_sat_host, activation_key=ak.name, org=org.label, rhel_distro=DISTRO_RHEL8
+    rhel_contenthost.configure_rex(satellite=rhcloud_sat_host, org=org, register=False)
+    rhel_contenthost.configure_rhai_client(
+        satellite=rhcloud_sat_host,
+        activation_key=ak.name,
+        org=org.label,
+        rhel_distro=distro_repo_map.get(rhel_contenthost.os_version.major),
     )
-    yield rhel8_contenthost_module
-
-
-@pytest.fixture
-def fixable_rhel8_vm(rhel8_insights_vm):
-    """A function-level fixture to create dnf related insights recommendation for rhel8 host."""
-    rhel8_insights_vm.run('dnf update -y dnf')
-    rhel8_insights_vm.run('sed -i -e "/^best/d" /etc/dnf/dnf.conf')
-    rhel8_insights_vm.run('insights-client')
+    rhel_contenthost.run('chmod 777 /etc/ssh/sshd_config;insights-client')
```

Comment:
@jameerpathan111 , will the `chmod 777 /etc/ssh/sshd_config;insights-client` not be enough for the rhel 8 client    
 to send the data ?

Reply:
@adarshdubey-star it'd be enough and that's what `test_rhcloud_insights_e2e` use. But for rest of the tests I am keeping the dnf rule, no particular reason. But if you want you can use openssh issue instead of dnf one later.